common/src/error.rs

```diff
@@ -15,6 +15,8 @@ pub enum Error {
     InvalidTimestamp,
```

Comment:
Perhaps we need to have a Display for each error variant for a better explanation on the logging